generators/chipyard/src/main/scala/config/AbstractConfig.scala

```diff
@@ -51,38 +57,68 @@ class AbstractConfig extends Config(
-  new freechips.rocketchip.subsystem.WithJtagDTM ++                 // set the debug module to expose a JTAG port
-  new freechips.rocketchip.subsystem.WithNoMMIOPort ++              // no top-level MMIO master port (overrides default set in rocketchip)
-  new freechips.rocketchip.subsystem.WithNoSlavePort ++             // no top-level MMIO slave port (overrides default set in rocketchip)
-  new freechips.rocketchip.subsystem.WithInclusiveCache ++          // use Sifive L2 cache
-  new freechips.rocketchip.subsystem.WithNExtTopInterrupts(0) ++    // no external interrupts
-  new freechips.rocketchip.subsystem.WithDontDriveBusClocksFromSBus ++ // leave the bus clocks undriven by sbus
-  new freechips.rocketchip.subsystem.WithCoherentBusTopology ++     // hierarchical buses including sbus/mbus/pbus/fbus/cbus/l2
-  new freechips.rocketchip.subsystem.WithDTS("ucb-bar,chipyard", Nil) ++ // custom device name for DTS
-  new freechips.rocketchip.system.BaseConfig)                       // "base" rocketchip system
+
+  // Peripheral section
+  new chipyard.config.WithUART ++                                   /** add a UART */
+  
+  // Core section
+  new chipyard.config.WithBootROM ++                                /** use default bootrom */
+  new testchipip.boot.WithCustomBootPin ++                          /** add a custom-boot-pin to support pin-driven boot address */
+  new testchipip.boot.WithBootAddrReg ++                            /** add a boot-addr-reg for configurable boot address */
+  
+  // ==================================
+  //   Set up tiles
+  // ==================================
+  // Debug settings
+  new chipyard.config.WithDebugModuleAbstractDataWords(8) ++        /** increase debug module data capacity */
+  // TODO: add these fragments
+  // new chipyard.config.WithJTAGDTMKey(idcodeVersion = 2, partNum = 0x000, manufId = 0x489, debugIdleCycles = 5) ++
+  // new freechips.rocketchip.subsystem.WithNBreakpoints(2) ++
+  new freechips.rocketchip.subsystem.WithJtagDTM ++                 /** set the debug module to expose a JTAG port */
+  
+  // Cache settings
+  new chipyard.config.WithL2TLBs(1024) ++                           /** use L2 TLBs */
+  new freechips.rocketchip.subsystem.WithInclusiveCache ++          /** use Sifive L2 cache */
+  
+  // Memory settings
+  new freechips.rocketchip.subsystem.WithNMemoryChannels(1) ++      /** Default 1 memory channels */
+  new freechips.rocketchip.subsystem.WithNoMMIOPort ++              /** no top-level MMIO master port (overrides default set in rocketchip) */
+  new freechips.rocketchip.subsystem.WithNoSlavePort ++             /** no top-level MMIO slave port (overrides default set in rocketchip) */
+  new freechips.rocketchip.subsystem.WithCoherentBusTopology ++     /** hierarchical buses including sbus/mbus/pbus/fbus/cbus/l2 */
+
+  // Core Settings
+  new freechips.rocketchip.subsystem.WithNExtTopInterrupts(0) ++    /** no external interrupts */
```

Comment:
Peripheral

Reply:
I'm not sure why this is peripheral. It's not on the memory address, right?